ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/KeyByBigQueryTableDestination.java

```diff
@@ -133,10 +133,16 @@ public TableDestination getTableDestination(Map<String, String> attributes) {
               msg = PubsubConstraints.ensureNonNull(msg);
               return KV.of(getTableDestination(msg.getAttributeMap()), msg);
             }).exceptionsInto(TypeDescriptor.of(PubsubMessage.class))
-            .exceptionsVia((WithFailures.ExceptionElement<PubsubMessage> ee) -> FailureMessage.of(
-                KeyByBigQueryTableDestination.class.getSimpleName(), //
-                ee.element(), //
-                ee.exception())));
+            .exceptionsVia((WithFailures.ExceptionElement<PubsubMessage> ee) -> {
+              if (ee.exception().getCause() instanceof ExecutionException
+                  || ee.exception().getCause() instanceof UncheckedIOException) {
```

Comment:
This is the best I could come up with to check for the exceptions that should be re-thrown. I wrapped checked exceptions into `RuntimeException` since it's not possible to throw checked exceptions.

Reply:
I'd prefer that we convert ExecutionException to UncheckedExecutionException:

```
     } catch (ExecutionException e) {
        throw new UncheckedExecutionException(e.getCause());
    }
```

Then here, we rethrow and catch only `IllegalArgumentException`. In general, I think we should only return a FailureMessage for a distinct list of exception types we expect rather than checking a specific list of exception types to bubble up to the pipeline level as we're doing here.